aws/amazon-ecs-cli
# feat(cicd): enable compilation through CodeBuild

Updated:
1. buildspec.yml so that the project can be built with CodeBuild.
2. Makefile so that version number, git dirty and git hash will be injected when compiled locally.

**THIS SHOULD NOT BE MERGED IN UNTIL OUR CI/CD INFRASTRUCTURE IS READY!**

**Testing**
- [N/A] Unit tests passed
- [N/A] Integration tests passed
- [N/A] Unit tests added for new functionality
- [V] Listed manual checks and their outputs in the comments ([example](https://github.com/aws/amazon-ecs-cli/pull/750#issuecomment-472623042))
- [N/A] Link to issue or PR for the integration tests:

**Documentation**
- [N/A] Contacted our doc writer
- [N/A] Updated our README

** Manual testing **
When the CLI is built locally with make:
```
./bin/local/ecs-cli --version
ecs-cli version v1.17.0 (*5972dd7)
```

When the CLI is through our new CI/CD:
```
./bin/local/ecs-cli --version
ecs-cli version 1.17.0 (10f9284)
```

The buildspec is verified with our CI/CD infrastructure.

----

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
